### PR TITLE
meson: inform ld to accept undefined symbols for version-scripts

### PR DESCRIPTION
**Pull request by bapt**

### CODE REVIEW

**Comment by whot** [comment]
Thanks. I copied the comment from #424 into your commit message so we have a bit more info here.

**Comment by glechapelain** [comment]
I'll verify here but this seems to be causing grieves to pkgsrc using GNU:

https://gnats.netbsd.org/cgi-bin/query-pr-single.pl?number=58272

File: meson.build [hunk 1]
```diff
@@ -145,7 +145,7 @@ configh_data.set('WIN32_LEAN_AND_MEAN', 1)
 # Supports -Wl,--version-script?
 have_version_script = cc.links(
     'int main(){}',
-    args: '-Wl,--version-script=' + meson.current_source_dir()/'xkbcommon.map',
+    args: '-Wl,--undefined-version,--version-script=' + meson.current_source_dir()/'xkbcommon.map',
```
Comment (eli-schwartz) [inline meson.build:148]: Certainly it is a bug if libxkbcommon depends on linker options that only work with specific bleeding-edge versions of binutils.

I am surprised that this broken PR was ever merged, at least in the state it was in. This is an lld-specific issue, so it should have been gated on `cc.get_linker_id() == 'lld'`.

Comment (wismill) [reply meson.build:148]: Let’s focus on this issue in #481.

**Pull request merged**